### PR TITLE
fixes printing non-contiguous tensors

### LINKED ISSUE #10402: Error on printing non-contiguous tensors
## Issue description

`get_summarized_data` in `_tensor_str.py` crashes when (unintentionally) a non-contiguous `tensor` is printed. 
The `.view(-1)` call in [this line of code](https://github.com/pytorch/pytorch/blob/e9ad74357e43d90f773abe31de88bd5f61db23eb/torch/_tensor_str.py#L216) throws an error, as `view(-1)` cannot be called on an non-contiguous `tensor`.

## Code example

```python
import torch
import torch.nn as nn

batch_size = 10
channels = 3
h, w = 16, 16
x = torch.randn(batch_size, channels, h, w)

pool = nn.AvgPool2d((1, 2), (1, 2))
output = pool(x)
output = output.view(batch_size, channels, h//2, 2, w//2)
print(output.select(3, 1).mul_(-1))
```

This code snippet crashes in the last line. I've noticed this issue while working in the Spyder IDE (REPL), where the line of code is also printed without the explicit `print` statement.

## System Info

Noticed this in all versions `> 0.4.0`, i.e. `0.4.1` and `0.5.0a0+1a0f4b5`.
[This commit](https://github.com/pytorch/pytorch/commit/90e75c65286b3799ebfd4e69d1ce0890519e581f#diff-f4eb2d7bbb99a6ea648fcc476f38659eR200) seems to be related to this issue.

## Workaround

We could call `self = self.contiguous()` before the line in question.
If `self` is already contiguous, this should result in a no-op.

I'll create a PR with this fix.

### PR DESCRIPTION
**Pull request by ptrblck**

### CODE REVIEW

**Comment by ptrblck** [comment]
Link to issue: https://github.com/pytorch/pytorch/issues/10402

**Comment by ezyang** [comment]
Can we get a test?

**Comment by ptrblck** [comment]
@ezyang Do you mean a specific test in `pytorch/test/..` or an executable code snippet?
If the latter:
```python
import torch
import torch.nn as nn

batch_size = 10
channels = 3
h, w = 16, 16
x = torch.randn(batch_size, channels, h, w)

pool = nn.AvgPool2d((1, 2), (1, 2))
output = pool(x)
output = output.view(batch_size, channels, h//2, 2, w//2)
print(output.select(3, 1).mul_(-1))
```
doesn't throw an error in the last line.

If the former, sure, where should I add it?

File: torch/_tensor_str.py [hunk 1]
```diff
@@ -213,6 +213,7 @@ def get_summarized_data(self):
         else:
             return self
     if self.size(0) > 2 * PRINT_OPTS.edgeitems:
+        self = self.contiguous()
```
Comment (SsnL) [inline torch/_tensor_str.py]: Should use `self = self.reshape(self.size(0), -1)` instead. See https://github.com/pytorch/pytorch/issues/10402#issuecomment-412174326

**Review by ssnl — DISMISSED** [review]

**Comment by SsnL** [comment]
@ptrblck For test, you can manually create a tensor and use `as_strided` to get the special size and strides, and add the test case to `TestTorch` (probably `test_print`).

File: torch/_tensor_str.py (hunk 1 above)
Comment (SsnL) [reply torch/_tensor_str.py]: Or better, just reshape the printed rows.

**Comment by ptrblck** [comment]
@ezyang @SsnL I've added a test case and used `.reshape(-1)` on the printed lines.
Somehow the tests are failing, so I'll have a look for the reason.

**Review by colesbury — APPROVED** [review]
lgtm

**Review by soumith — APPROVED** [review]

**Pull request closed without merging**